src/pages/storage/StorageVolumeForm.tsx

```diff
@@ -0,0 +1,298 @@
+  }
+
+  if (!pool) {
+    return <>Missing storage pool name</>;
+  }
+
+  const { data: storagePools = [], error } = useQuery({
+    queryKey: [queryKeys.storage],
+    queryFn: () => fetchStoragePools(project),
+  });
+
+  if (error) {
+    notify.failure("Loading storage pools failed", error);
+  }
+
+  const StorageSchema = Yup.object().shape({
+    name: Yup.string()
+      .test(...testDuplicateName(project, controllerState))
+      .required("This field is required"),
+  });
+
+  const formik = useFormik<StorageVolumeFormValues>({
+    initialValues: {
+      name: "",
+      pool: pool,
+      size: "",
+    },
+    validationSchema: StorageSchema,
+    onSubmit: (values) => {
+      const volume = {
+        name: values.name,
+        config: {
+          size: values.size ? `${values.size}GiB` : undefined,
+          "security.shifted": toBool(values.security_shifted),
+          "security.unmapped": toBool(values.security_unmapped),
+          "snapshots.expiry": values.snapshots_expiry,
+          "snapshots.pattern": values.snapshots_pattern,
+          "snapshots.schedule": values.snapshots_schedule,
+          "block.filesystem": values.block_filesystem,
+          "block.mount_options": values.block_mount_options,
+          "lvm.stripes": values.lvm_stripes,
+          "lvm.stripes_size": values.lvm_stripes_size,
+          "zfs.blocksize": values.zfs_blocksize,
+          "zfs.block_mode": toBool(values.zfs_block_mode),
+          "zfs.delegate": toBool(values.zfs_delegate),
+          "zfs.remove_snapshots": toBool(values.zfs_remove_snapshots),
+          "zfs.use_refquota": toBool(values.zfs_use_refquota),
+          "zfs.reserve_space": toBool(values.zfs_reserve_space),
+        },
+        project: project,
+        type: "custom",
+        content_type: "filesystem",
+      };
+
+      createStorageVolume(values.pool, project, volume)
+        .then(() => {
+          void queryClient.invalidateQueries({
+            queryKey: [queryKeys.storage],
+          });
+          navigate(
+            `/ui/project/${project}/storage/detail/${values.pool}/volumes`,
+            notify.queue(
+              notify.success(`Storage volume ${values.name} created.`)
+            )
+          );
+        })
+        .catch((e) => {
+          formik.setSubmitting(false);
+          notify.failure("Storage volume creation failed", e);
+        });
+    },
+  });
+
+  const submitForm = () => {
+    void formik.submitForm();
+  };
+  const getFormProps = (id: keyof StorageVolumeFormValues) => {
+    return {
+      id: id,
+      name: id,
+      onBlur: formik.handleBlur,
+      onChange: formik.handleChange,
+      value: formik.values[id] ?? "",
+      error: formik.touched[id] ? (formik.errors[id] as ReactNode) : null,
+      stacked: true,
+    };
+  };
+
+  const poolDriver =
+    storagePools.find((item) => item.name === pool)?.driver ?? "";
+
+  return (
+    <BaseLayout title="Create volume">
+      <NotificationRow />
```

Comment:
Maybe the notification area should stick to the top and not be scrolled, so that I don't have to manually scroll to the top to see the notification when there I get one?